### ISSUE TITLE
Android - Home Screen - Poor Focus Ring/Border when TalkBack is Reading

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#18]
When reading with TalkBack, I’m having trouble noticing the light blue focus ring when it gets to the blue and green sections on the home screen. So my suggestion is to change it to black ( #000000 ), so it can be seen and has enough color contrast.

Attempted to take a screenshot and it does not show the TalkBack focus ring.

**Comment by KyleLivingstonDOL** [comment]
Thanks John. Noted.

**Comment by KyleLivingstonDOL** [comment]
Hi John,

What is the focus ring color contrast needed? Is it 4.5:1, like other contrasts?

-Kyle

**Comment by jfc3-dol** [comment]
So keyboard users or those using a screen reader to read along can see where on a page/screen they are located.

Yes, the outline ring needs to have enough color contrast between itself and the background color along with itself, and in this example, the button. In my years of programming, I have found that inverting the color of the button and text color can work the best. An example would be if the text is white and the background of the button is dark blue when it in its normal state that if you make the text and border color the same blue with a white background, you should be fine unless the button is on top of another background color that won't work.

I use Jonathan Snook's online Colour Contrast Checker ( https://snook.ca/technical/colour_contrast/colour.html ) to make sure the color contrast ratio is 4.5:1, and both of the WCAG 2.0 AA boxes say "YES". All you have to do is put the text and background colors in the foreground and background fields, and it will tell you what you need to know.

**Comment by KyleLivingstonDOL** [comment]
Thanks John, we can do this.
ZTI, can we make the focus rings a stark contrast? Like red?

**Comment by LeHaine** [comment]
https://github.com/USDepartmentofLabor/Comply-Chain/issues/37#issuecomment-498685874

**Comment by jfc3-dol** [comment]
The TalkBack blue focus ring does not have enough color contrast when focusing on the text over the colored images.

If this is an Android operating system issue I would like to keep this open until there is a fix since you said CSS changes aren't working.

**Comment by KyleLivingstonDOL** [comment]
John,

Thanks for this. We cannot change the Talkback focus ring color within Comply Chain as the ring is not part of Comply Chain. Rather the ring's color is part of Talkback's program. As seen above, your ring is blue but Colton's ring is green. We understand that neither have the require contrast to sufficiently address the 508 issue. Since changing the button colors would require us to go back to the beginning of app development to change the entire color scheme, we would like to put this issue on hold for now (along with #37) until a time comes when we can influence the Talkback focus ring color from Comply Chain.

Best,
Kyle

**Comment by jfc3-dol** [comment]
At this point, it makes sense to put this and issue #37 on hold since the number of sighted TalkBack users that are also color deficient (colorblind) is probably that huge.

**Comment by KyleLivingstonDOL** [comment]
Thanks John. Confirming we will put this on hold for now (and #37)
-Kyle

**Comment by jfc3-dol** [comment]
At this point, it makes sense to put this and issue #37 on hold since the number of sighted TalkBack users that are also color deficient (colorblind) is probably that huge.

**Comment by jfc3-dol** [comment]
If that is the case with this issue or any other that is not left open and not accessible OPA sends out a document listing the number of issues found, still open, and requests at least an e-mail stating the given agency is willing to take the risk if someone complains in whatever way that it's not accessible. Which has been done by all of them as far as I know.

**Comment by KyleLivingstonDOL** [comment]
Ok. Thanks John. I appreciate this information. Do you know what the procedure is to start this process?

**Comment by jfc3-dol** [comment]
Once we are done testing the entire Comply Chian application (iOS, Android, and web) I create and send to you and my boss OPAs findings of the number of issues found; the number closed, etc. At which point we then look for a signature (can be electronic or an e-mail) from the agency that they are willing to take the risk of having known accessibility issues when moving everything to PROD.

By documenting things in GitHub with all these types of comments, we could explain to someone that some of them are platform issues, etc. OPA doesn't want to put an agency or DOL at risk for complaints, etc. from people along with we are working on them.

**Comment by jfc3-dol** [comment]
Marking this issue for a "Future release" by using the label "future release" so we all know we are most likely not finishing it this release unless we manage to figure out how to do it.

**Comment by jfc3-dol** [comment]
It still seems to be a problem, so the color contrast can't be improved on Android.

**Comment by pbhatt17** [comment]
@jfc3-dol 
This ring is from talkback and can't modify. Do you still want to keep this ticket open?

**Comment by jfc3-dol** [comment]
@pbhatt17 

I'm keeping this open to remind us that the TalkBack focus ring still has poor color contrast. It probably won't close until Android allows developers to change the color of the focus ring when using TalkBack or a blue tooth keyboard.
